### ISSUE TITLE
Custom exception fallback in documentation does not compile

### ISSUE THREAD

**Issue by eduanb** [jhalterman/failsafe#193]
The example in the readme for a  custom fallback exception: 

```java
Fallback<Object> fallback = Fallback.of(failure -> { throw new CustomException(failure); });
```

Gives the following compiler exception:
`ERROR: reference to of is ambiguous
  both method <R>of(net.jodah.failsafe.function.CheckedConsumer<net.jodah.failsafe.event.ExecutionAttemptedEvent<? extends R>>) in net.jodah.failsafe.Fallback and method <R>of(net.jodah.failsafe.function.CheckedFunction<net.jodah.failsafe.event.ExecutionAttemptedEvent<? extends R>,? extends R>) in net.jodah.failsafe.Fallback match`

I've got it to compile if I define them seperately like this:
```java
CheckedFunction<ExecutionAttemptedEvent, Boolean> function = failure -> { throw new CustomException(failure); };
		Fallback fallback = Fallback.of(function);
```
Don't think this affects it, but I am using `openjdk version "11.0.3"`.

**Comment by jhalterman** [comment]
Thanks for filing this.

This ambiguity seems to only occur when we want to fallback to an alternate exception, so I think the best way to deal with this is to create a specific method. So I added `Fallback.ofException` for this and updated the README.

**Comment by jhalterman** [comment]
Released in 2.1.1
